### ISSUE TITLE
General vault counts don't correctly count mods and transmat effects

### ISSUE THREAD

**Issue by thezoggy** [DestinyItemManager/DIM#2245]
mod count is not correct
![dim](http://zoggy.net/firefox_2017-09-26_11-15-20.png)

~I can only assume that the item stack amount is being used rather than just the amount of items?~

looking at what DIM shows.. there is 55 items:

![dim2](http://zoggy.net/firefox_2017-09-26_11-18-02.png)

there is 5  Transmat Effects.. wonder if these are whats screwy?

**Comment by bhollis** [comment]
How do those show up in the game?

**Comment by kyleshay** [comment]
Possibly related to #2245

**Comment by thezoggy** [comment]
@kyleshay I believe you linked to the wrong GI.
@bhollis Let me pull up the game and check...

**Comment by thezoggy** [comment]
TL;DR; DIM shows 'user' limits for character vault not global vault limits.

Logged into a character and started to remove from the vault all the legendary mods until my character vault is full (50/50) to try and simplify the amounts.
 
In the game, my vault is `16/200`.
Which currently consist of 6 weapons and the rest being mods/exotic ornaments/transmat effects.

DIM shows this just split, with the weapons up top and the non-weapons on the bottom.
![dim_vault](http://zoggy.net/firefox_2017-09-28_21-52-51.png)
![dim_vault2](http://zoggy.net/firefox_2017-09-28_21-56-03.png)

DIM just then does funky math up top showing that the general vault is `6/200` as it is filtering to just weapons. Then shows mods is `10/50`. If you combine these.. 16 is the actual total of items in the vault. The max amount in the vault is 200 (for weapons+mods). 

Now on the actual character itself, you can only have 50 mods.
You can have 200 in the vault.

for example I moded all my mods to the vault, shows `66/200` (6 are weapon, 60 are non-weapon):
![dim_vault3](http://zoggy.net/firefox_2017-09-28_22-06-24.png)

So it looks like the actual per char vault vaules just need to be added.. maybe just add under the power level part as another section.. then the 50 limit would actually be valid.

**Comment by DaoFerret** [comment]
@bhollis just checked on 4.18.0  The Consumables, Shaders and Modifications counts are correct (thank you for all your hard work).  The General vault count though, is not including the count of Mods and Shaders that are in the vault (so its displayed count is lower than reality).

**Comment by bhollis** [comment]
Hmm, I’m sure I checked that. Let me try again. 

-Ben

**Comment by bhollis** [comment]
Ah, you're right. I swore I'd checked that. Let me give it another pass. The way we handle vaulted inventory (splitting it up into groups instead of one big unsorted bucket) unfortunately makes this tricky. We'll have to have a better structure for the new DIM.

**Comment by bhollis** [comment]
Hmm I went back and reloaded and now it's working great, beta and app. Did I dream this?

**Comment by duckmanBR** [comment]
Right now in beta, C/S/M are showing the values for my char (I only have one) and not on Vault...

**Comment by crist074** [comment]
I am still having issues with incorrect vault counts as well.

**Comment by bhollis** [comment]
To be clear, there is no separate vault for consumables, shaders, and mods,
so those should get counted based on the number in the account-wide
buckets. When you move these to the vault, you should see the "general"
bucket count up, since there's only one shared vault. I'm sorry this is
confusing, but the new design of the vault makes it a bit odd to display.

**Comment by kyleshay** [comment]
Would it be less confusing if it looked like this?

<img width="1010" alt="screen shot 2017-10-03 at 7 46 15 pm" src="https://user-images.githubusercontent.com/424158/31158002-948efc44-a873-11e7-9605-cf4e38911007.png">

If we did do this, it would set us up pretty nicely for showing the postmaster and engram counts as well.

**Comment by crist074** [comment]
I have no problem with the C/S/M numbers. The issue I have is that the general counter in DIM does not reflect what my actual vault says. I noticed that my actual in game vault count minus my DIM general counter equals the number of C/S/M in my vault i.e. DIM is not counting those items as part of the general.

**Comment by duckmanBR** [comment]
@bhollis the problem is that you are removing from G the count of C/S/M when you do the first reading of the vault (after open/refresh).

Example:
My vault is full right now, 200 items split in 113 W/A/G/Sp/Sh and 87 C/S/M (including transmats and ornaments)
If I open/refresh DIM under my vault will read: 
G 113/200 C 19/50 S 0/50 M 0/50
Then I'll remove one of each C/S/M from vault:
G 110/200 C 20/50 S 1/50 M 1/50
But if I refresh the app:
G 113/200 C 20/50 S 1/50 M 1/50

If C/S/M are values not related to the vault, they shouldn't be removed.

Anyway, this should be revised.  @kyleshay idea is better than what we have, but then a break in how your vault is filled would be nice, but (W and A would also be welcome)

**Comment by bhollis** [comment]
Yeah, I think I understand the bug, and I can fix it. The vault stuff got complicated because we like to show vault broken out by type instead of all together. 

I'm down to change the display though just showing one number u set the vault Seema like a waste of space. 

-Ben
